src/material-experimental/mdc-slider/global-change-and-input-listener.ts

```diff
@@ -7,64 +7,61 @@
+ * #fromEvent which will lazily attach a listener when the first subscription is made and remove the
+ * listener once the last observer unsubscribes.
  */
 @Injectable({providedIn: 'root'})
-export class GlobalChangeAndInputListener<K extends 'change'|'input'> {
+export class GlobalChangeAndInputListener<K extends 'change'|'input'> implements OnDestroy {
 
   /** The injected document if available or fallback to the global document reference. */
   private _document: Document;
 
   /** Stores the subjects that emit the events that occur on the global document. */
-  private _subjects = new Map<K, Subject<Event>>();
+  private _observables = new Map<K, Observable<Event>>();
 
-  /** Stores the event handlers that emit the events that occur on the global document. */
-  private _handlers = new Map<K, ((event: Event) => void)>();
+  /** The notifier that triggers the global event observables to stop emitting and complete. */
+  private _destroyed = new Subject();
 
   constructor(@Inject(DOCUMENT) document: any, private _ngZone: NgZone) {
     this._document = document;
   }
 
-  /** Returns a function for handling the given type of event. */
-  private _createHandlerFn(type: K): ((event: Event) => void) {
-    return (event: Event) => {
-      this._subjects.get(type)!.next(event);
-    };
+  ngOnDestroy() {
+    this._destroyed.next();
+    this._destroyed.complete();
+    this._observables.clear();
   }
 
   /** Returns a subscription to global change or input events. */
   listen(type: K, callback: SpecificEventListener<K>): Subscription {
-    // This is the first subscription to these events.
-    if (!this._subjects.get(type)) {
-      const handlerFn = this._createHandlerFn(type).bind(this);
-      this._subjects.set(type, new Subject<Event>());
-      this._handlers.set(type, handlerFn);
-      this._ngZone.runOutsideAngular(() => {
-        this._document.addEventListener(type, handlerFn, true);
-      });
+    // If this is the first time we are listening to this event, create the observable for it.
+    if (!this._observables.has(type)) {
+      this._observables.set(type, this._createGlobalEventObservable(type));
     }
 
-    const subject = this._subjects.get(type)!;
-    const handler = this._handlers.get(type)!;
+    return this._ngZone.runOutsideAngular(() =>
+      this._observables.get(type)!.subscribe((event: Event) =>
+        this._ngZone.run(() => callback(event))
+      )
+    );
+  }
 
-    return subject.pipe(finalize(() => {
-        // This is the last event listener unsubscribing.
-        if (subject.observers.length === 1) {
-          this._document.removeEventListener(type, handler, true);
-          this._subjects.delete(type);
-          this._handlers.delete(type);
-        }
-    })).subscribe(callback);
+  /** Creates an observable that emits all events of the given type. */
+  private _createGlobalEventObservable(type: K) {
+    return fromEvent(this._document, type, {capture: true}).pipe(
```

Comment:
Now that I think about it we should also make this `passive: true` as long as we're not calling `preventDefault` (IIRC we're just calling `stopPropagation`)